app/models/history.rb

```diff
@@ -108,6 +109,37 @@ class History < ApplicationRecord
       .where(history_type: [HISTORY_TYPES[:reports_reviewed], HISTORY_TYPES[:report_reviewed]])
   }
 
+  def self.errored_report_reminder_group_of_patients(patients: nil, created_by: 'Sara Alert System', comment: 'Failed Contact Attempt', error_message: nil)
+    histories = []
+    patients.uniq.each do |pat|
+      # Avoid adding multiple errored history items if HoH has errored contact attempts for
+      # contact attempts to them on behalf of their dependents. This would be the
+      # case for failed weblinks
+      last_history = pat&.histories&.last
+      next if !last_history.nil? && (last_history.history_type == 'Unsuccessful Report Reminder') && ((Time.now - last_history&.created_at) / 60) < 5
```

Comment:
I'd like to get everyone's opinion about this -- The idea is to keep many of the same exact error history items getting reported to a HoH for each of their dependents. Would it be better to have all the errored history items and run the risk of different history items on the HoH get buried beneath a series of Unsuccessful report reminders for each of their dependents? Or is it better to have a history item detailing the failure for each dependent?

Reply:
The more I think about this, I'm not so sure it's the best approach here. I do think grouping history items is something I'd like to do. But if I was user, I think I'd want to know for weblink if any of the dependent weblinks were successfully sent or not. Given we have the filtering functionality I'm not too worried about history getting buried personally.

Reply:
From our typical error rate, can we extrapolate how many history elements this would actually save?

Reply:
Yes and No -- Yes we could get the rough order of magnitude but No in that this PR _also_ addresses 1121 which will cause history items to be added in cases they currently are not eg: failed dependents whose HoH is not active <- these cases would not be captured by which ever queries were run to get the rough order of magnitude